lens/util/repo.go

```diff
@@ -31,11 +30,55 @@ import (
 var ActorRegistry *vm.ActorRegistry
 
 func init() {
-	ActorRegistry = filcns.NewActorRegistry()
+	ActorRegistry = consensus.NewActorRegistry()
 }
 
 var log = logging.Logger("lily/lens")
 
+func ParseVmMessageParams(params []byte, paramsCodec uint64, method abi.MethodNum, actCode cid.Cid) (string, string, error) {
+	m, found := ActorRegistry.Methods[actCode][method]
+	if !found {
+		// if the method wasn't found it is likely the case the method was one of:
+		// https://github.com/filecoin-project/builtin-actors/blob/f5311fe735df4d9baf5f82d4b3db10f3c51688c4/actors/docs/README.md?plain=1#L31
+		// so we just marshal the raw value to json and bail with a warning
+		paramj, err := json.Marshal(params)
+		if err != nil {
+			return "", "", err
+		}
+		err = fmt.Errorf("unknown method %d with codec %d for actorCode %s name %s", method, paramsCodec, actCode, builtin.ActorNameByCode(actCode))
+		log.Warnw("parsing vm message params", "error", err)
+		return string(paramj), builtin.ActorNameByCode(actCode), nil
+	}
+	// If the codec is 0, the parameters/return value are "empty".
+	// If the codec is 0x55, it's bytes.
+	if paramsCodec == 0 || paramsCodec == 0x55 {
+		return string(params), m.Name, nil
+	}
+	return ParseParams(params, method, actCode)
+}
+
+func ParseVmMessageReturn(ret []byte, retCodec uint64, method abi.MethodNum, actCode cid.Cid) (string, string, error) {
+	m, found := ActorRegistry.Methods[actCode][method]
+	if !found {
+		// if the method wasn't found it is likely the case the method was one of:
+		// https://github.com/filecoin-project/builtin-actors/blob/f5311fe735df4d9baf5f82d4b3db10f3c51688c4/actors/docs/README.md?plain=1#L31
+		// so we just marshal the raw value to json and bail with a warning
+		retJ, err := json.Marshal(ret)
+		if err != nil {
+			return "", "", err
+		}
+		err = fmt.Errorf("unknown method %d with codec %d for actorCode %s name %s", method, retCodec, actCode, builtin.ActorNameByCode(actCode))
+		log.Warnw("parsing vm message return", "error", err)
+		return string(retJ), builtin.ActorNameByCode(actCode), nil
+	}
+	// If the codec is 0, the parameters/return value are "empty".
+	// If the codec is 0x55, it's bytes.
+	if retCodec == 0 || retCodec == 0x55 {
+		return string(ret), m.Name, nil
+	}
+	return ParseReturn(ret, method, actCode)
+}
+
```

Comment:
Special handling for vm message params and returns please review this section, other changes are mechanical